android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java

```diff
@@ -163,6 +166,63 @@ public void run() {
         });
     }
 
+    @ReactMethod
+    public void getCurrentMedia(final Promise promise) {
+      if (mCastSession != null) {
+        getReactApplicationContext().runOnUiQueueThread(new Runnable() {
+          @Override
+          public void run() {
+            RemoteMediaClient client = mCastSession.getRemoteMediaClient();
+            if (client == null) {
+              return;
+            }
+            try {
+              if (client.getCurrentItem() == null) {
+                promise.resolve(null);
+                return;
+              }
+              MediaInfo media = client.getCurrentItem().getMedia();
+              WritableMap params = Arguments.createMap();
+              params.putString("title", media.getMetadata().getString(MediaMetadata.KEY_TITLE));
+              String subtitle = media.getMetadata().getString(MediaMetadata.KEY_SUBTITLE);
+              if (subtitle != null) {
+                params.putString("subtitle", media.getMetadata().getString(MediaMetadata.KEY_SUBTITLE));
+              }
+              params.putString("mediaUrl", media.getContentId());
+              JSONObject customData = media.getCustomData();
+              if (customData != null) {
+                params.putMap("customData", WritableMapUtils.toWritableMap(customData));
+              }
+              if (media.getMetadata().getImages() != null && media.getMetadata().getImages().size() > 0) {
+                  WebImage image = media.getMetadata().getImages().get(0);
+                  params.putString("imageUrl", image.getUrl().toString());
+              }
+              promise.resolve(params);
+            } catch (Exception e) {
+              promise.reject(e);
+            }
+
+          }
+        });
+      }
+    }
+
+    @ReactMethod
+    public void setPlaybackRate(final double rate) {
+      if (mCastSession != null) {
+        getReactApplicationContext().runOnUiQueueThread(new Runnable() {
+          @Override
+          public void run() {
```

Comment:
Is there any possibility of there being a race condition here where by the time the Runnable runs on the Ui thread, that mCastSession is already gone?  What happens if this code `throw`s? Should there be a catch inside the `run()` handler to avoid bombing the app in error cases?